src/user.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import json
+import uuid
+from datetime import datetime
+import logging
+from google.auth import jwt
+from flask import Blueprint, jsonify, request
+from src.constants.http_status_codes import HTTP_200_OK,HTTP_201_CREATED, HTTP_400_BAD_REQUEST, HTTP_401_UNAUTHORIZED
+from src.models import UserLogin, UserProfile, db, Borrowing
+
+
+user = Blueprint('user', __name__, url_prefix='/api/v1/user')
+
+
+def decode_token(token_object):
+     
+    """
+    Decodes a google token and returns a JSON object with the following details
+    {
+        "at_hash": "ekLIO2gVVwjiH02eiu88hQ",
+        "aud": "170044535313-f02c8hd020ptha7n0t4fraouo2ap9bqq.apps.googleusercontent.com",
+        "azp": "170044535313-f02c8hd020ptha7n0t4fraouo2ap9bqq.apps.googleusercontent.com",
+        "email": "email@mail.com",
+        "email_verified": true,
+        "exp": 1636222353,
+        "family_name": "Last_Name",
+        "given_name": "First_Name",
+        "iat": 1000000000,
+        "iss": "accounts.google.com",
+        "jti": "0019b0aaa2e355f8fa2b2ce8b3bbbbab7b63a5014",
+        "locale": "en-GB",
+        "name": "First_Name Name",
+        "picture": "https://lh3.googleusercontent.com/a-/image_url..",
+        "sub": "113501893650341726537"
+    }
+    """
+    # try:
+    #     return jwt.decode(token_object, verify=False)
+    # except Exception as ex:
+    #     logging.critical(ex)
+    return jwt.decode(token_object, verify=False)
+
+
+@user.post('/login/google')
+def login():
+    if request.method != 'POST':
+        return HTTP_405_METHOD_NOT_ALLOWED
+    
+    # Checks on request.
+    if 'id_token' not in request.json:
+        return jsonify({
+            'error': "'id_token' is missing from request"
+        }), HTTP_400_BAD_REQUEST
+
+    token = request.json['id_token']
+    
+    try:
+        google_response = decode_token(token)
+    except Exception:
+        return {"error":"Invalid Token"}, HTTP_400_BAD_REQUEST
+
+    email = google_response['email']
+    first_name = google_response['given_name']
+    last_name = google_response['family_name']
+    picture_url = google_response['picture']
```

Comment:
What happens if tomorrow, Google changes the structure of the response, and these fields don't exist anymore? Better to handle edge cases alongside since we do not have control of the data coming in.